### PR TITLE
Check in Cloud Job Discovery client code samples for python

### PR DESCRIPTION
**Pull request by xinyunh0929**
CJD got Public beta launch so we would like to have our samples checked in as other projects.

### CODE REVIEW

**Comment by xinyunh0929** [comment]
I am working on the tests to those samples, however, I don't know how to run those '_test' files. Is there any guide?

File: jobs/auto_complete_sample.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+# [START instantiate]
+from googleapiclient.discovery import build
+
+client_service = build('jobs', 'v2')
+
```
Comment (michaelawyu) [inline jobs/auto_complete_sample.py]: You can remove the empty lines between code and region tag.

File: jobs/base_company_sample.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+# [START instantiate]
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+client_service = build('jobs', 'v2')
+
+# [END instantiate]
+
+
+# [START basic_company]
```
Comment (michaelawyu) [inline jobs/base_company_sample.py]: Maybe it's not necessary to declare a function here with a specific region tag. How about just specifying a `company` variable? 

```
 company = {
       'display_name': 'Google',
       'distributor_company_id': distributor_company_id,
       'hq_location': '1600 Amphitheatre Parkway Mountain View, CA 94043'
}
```

You can use a fixed ID here as well I think.

File: jobs/auto_complete_sample.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# [START auto_complete_job_title]
+def job_title_auto_complete(client_service, query, company_name):
+    complete = client_service.v2().complete(
+        query=query, languageCode='en-US', type='JOB_TITLE', pageSize=10)
+    if company_name is not None:
+        complete.companyName = company_name
+
+    results = complete.execute()
+    print('==========\n%s\n==========' % results)
```
Comment (michaelawyu) [inline jobs/auto_complete_sample.py]: The padding part `==========` may not be necessary.

File: jobs/base_company_sample.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+
+# [START create_company]
+def create_company(client_service, company_to_be_created):
+    try:
+        company_created = client_service.companies().create(
+            body=company_to_be_created).execute()
+        print('==========\nCompany created: %s\n==========' % company_created)
+        return company_created
+    except HttpError as e:
```
Comment (michaelawyu) [inline jobs/base_company_sample.py]: Use the base error class would be better I think.

File: jobs/base_job_sample.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+client_service = build('jobs', 'v2')
+
+# [END instantiate]
+
+
+# [START basic_job]
+def generate_job_with_required_fields(company_name):
```
Comment (michaelawyu) [inline jobs/base_job_sample.py]: I feel that this can be a simple `job` variable as well.

File: jobs/batch_operation_sample.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+from googleapiclient.discovery import build
+
+client_service = build('jobs', 'v2')
+
+# [END instantiate]
+
+
+# [START batch_job_create]
+def batch_job_create(client_service, company_name):
+    import base_job_sample
```
Comment (michaelawyu) [inline jobs/batch_operation_sample.py]: I understand your concern here; but for clarity I do feel that this part can be replaced with two simple `job` variables so that developers do not need to read another file to understand what's going on behind the scenes.

File: jobs/batch_operation_sample.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            print('==========\nJob updated: %s\n==========' % response)
+            updated_jobs.append(response)
+            pass
+
+    batch = client_service.new_batch_http_request()
+    for index in range(0, len(jobs_to_be_updated)):
+        job_to_be_updated = jobs_to_be_updated[index]
+        job_to_be_updated.update({'job_title': 'Engineer in Mountain View'})
+        request = {'job': job_to_be_updated}
+        if index % 2 == 0:
```
Comment (michaelawyu) [inline jobs/batch_operation_sample.py]: This part might be a little bit confusing I think. If we are showing the batch updating method do we really need to showcase updating only half of the jobs and leaving the other half alone? We can just update all of them.

File: jobs/batch_operation_sample.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        if exception is not None:
+            print('Got exception while updating job: %s' % exception)
+            pass
+        else:
+            print('==========\nJob updated: %s\n==========' % response)
+            updated_jobs.append(response)
+            pass
+
+    batch = client_service.new_batch_http_request()
+    for index in range(0, len(jobs_to_be_updated)):
```
Comment (michaelawyu) [inline jobs/batch_operation_sample.py]: Side note: if you switch to updating all jobs instead, this part can become `for job in jobs`, which is easier to understand and more idiomatic for Python devs.

File: jobs/batch_operation_sample.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    def job_delete_callback(request_id, response, exception):
+        if exception is not None:
+            print('Got exception while deleting job: %s' % exception)
+            pass
+        else:
+            print('==========\nJob deleted\n==========')
+            pass
+
+    batch = client_service.new_batch_http_request()
+    for index in range(0, len(jobs_to_be_deleted)):
```
Comment (michaelawyu) [inline jobs/batch_operation_sample.py]: The `index` is not necessary though; how about `for job in jobs`?

File: jobs/custom_attribute_sample.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+# [START instantiate]
+from googleapiclient.discovery import build
+
+client_service = build('jobs', 'v2')
+
+# [END instantiate]
+
+
+# [START custom_attribute_job]
+def generate_job_with_custom_attributes(company_name):
```
Comment (michaelawyu) [inline jobs/custom_attribute_sample.py]: Similarly, a variable would be better.

File: jobs/env/bin/activate.csh [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# This file must be used with "source bin/activate.csh" *from csh*.
```
Comment (michaelawyu) [inline jobs/env/bin/activate.csh]: Is this file necessary?

File: jobs/env/bin/activate [hunk 12]
```diff
@@ -0,0 +1,78 @@
+# This file must be used with "source bin/activate" *from bash*
```
Comment (michaelawyu) [inline jobs/env/bin/activate]: Please remove this file from the commit.

File: jobs/env/bin/activate.fish [hunk 13]
```diff
@@ -0,0 +1,76 @@
+# This file must be used using `. bin/activate.fish` *within a running fish ( http://fishshell.com ) session*.
```
Comment (michaelawyu) [inline jobs/env/bin/activate.fish]: Please remove this file from the commit.

File: jobs/env/bin/activate_this.py [hunk 14]
```diff
@@ -0,0 +1,34 @@
+"""By using execfile(this_file, dict(__file__=this_file)) you will
```
Comment (michaelawyu) [inline jobs/env/bin/activate_this.py]: Please remove this file from the commit.

File: jobs/env/bin/easy_install [hunk 15]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/easy_install]: Please remove this file from the commit.

File: jobs/env/bin/easy_install-2.7 [hunk 16]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/easy_install-2.7]: Please remove this file from the commit.

File: jobs/env/bin/pbr [hunk 17]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pbr]: Please remove this file from the commit.

File: jobs/env/bin/pip [hunk 18]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pip]: Please remove this file from the commit.

File: jobs/env/bin/pip2 [hunk 19]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pip2]: Please remove this file from the commit.

File: jobs/env/bin/pip2.7 [hunk 20]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pip2.7]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-decrypt [hunk 21]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-decrypt]: Please remove this file from the commit.

File: jobs/env/pip-selfcheck.json [hunk 22]
```diff
@@ -0,0 +1 @@
+{"last_check":"2018-07-16T22:25:20Z","pypi_version":"10.0.1"}
```
Comment (michaelawyu) [inline jobs/env/pip-selfcheck.json]: Please remove this file from the commit.

File: jobs/env/local/include [hunk 23]
```diff
@@ -0,0 +1 @@
+/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/include
```
Comment (michaelawyu) [inline jobs/env/local/include]: Please remove this file from the commit.

File: jobs/env/local/bin [hunk 24]
```diff
@@ -0,0 +1 @@
+/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin
```
Comment (michaelawyu) [inline jobs/env/local/bin]: Please remove this file from the commit.

File: jobs/env/include/python2.7 [hunk 25]
```diff
@@ -0,0 +1 @@
+/usr/include/python2.7
```
Comment (michaelawyu) [inline jobs/env/include/python2.7]: Please remove this file from the commit.

File: jobs/env/bin/wheel [hunk 26]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/wheel]: Please remove this file from the commit.

File: jobs/env/bin/python2.7 [hunk 27]
```diff
@@ -0,0 +1 @@
+python
```
Comment (michaelawyu) [inline jobs/env/bin/python2.7]: Please remove this file from the commit.

File: jobs/env/bin/python2 [hunk 28]
```diff
@@ -0,0 +1 @@
+python
```
Comment (michaelawyu) [inline jobs/env/bin/python2]: Please remove this file from the commit.

File: jobs/env/bin/python-config [hunk 29]
```diff
@@ -0,0 +1,78 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/python-config]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-verify [hunk 30]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-verify]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-sign [hunk 31]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-sign]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-priv2pub [hunk 32]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-priv2pub]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-keygen [hunk 33]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-keygen]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-encrypt-bigfile [hunk 34]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-encrypt-bigfile]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-encrypt [hunk 35]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-encrypt]: Please remove this file from the commit.

File: jobs/env/bin/pyrsa-decrypt-bigfile [hunk 36]
```diff
@@ -0,0 +1,11 @@
+#!/usr/local/google/home/xinyunh/githubCode/pythonPersonalFork/python-docs-samples/jobs/env/bin/python
```
Comment (michaelawyu) [inline jobs/env/bin/pyrsa-decrypt-bigfile]: Please remove this file from the commit.

File: jobs/auto_complete_sample.py (hunk 1 above)
Comment (xinyunh0929) [reply jobs/auto_complete_sample.py]: I use `pyformat -i -s 4 --recursive . ` to format my code. And it seems this is recommended. 
Should I manually modify those files? Or it should be fine?

File: jobs/auto_complete_sample.py [hunk 37]
```diff
@@ -0,0 +1,81 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+# [START instantiate]
```
[the remark this answers is not in the document]
Comment (xinyunh0929) [reply jobs/auto_complete_sample.py]: This is for tag consistence among different languages since TW prefer them in the same name. Do you think it is necessary to modify the tag?

File: jobs/base_company_sample.py (hunk 2 above)
Comment (xinyunh0929) [reply jobs/base_company_sample.py]: I could not use a fixed ID since creating an entity with same ID would lead to the failure of tests

Comment (xinyunh0929) [reply jobs/base_company_sample.py]: No. This function is required from doc site and I couldn't use the fixed ID due to our system.

File: jobs/base_company_sample.py (hunk 4 above)
Comment (xinyunh0929) [reply jobs/base_company_sample.py]: Done.

File: jobs/base_job_sample.py (hunk 5 above)
Comment (xinyunh0929) [reply jobs/base_job_sample.py]: Just to distinguish this function from other generate job function such as generate_job_with_custom_attribute

File: jobs/batch_operation_sample.py (hunk 6 above)
Comment (xinyunh0929) [reply jobs/batch_operation_sample.py]: It could be changed later since it request modifying the samples in all languages.

File: jobs/batch_operation_sample.py (hunk 7 above)
Comment (xinyunh0929) [reply jobs/batch_operation_sample.py]: They want to have the updateRequest one with the field mask feature, one is not.

File: jobs/batch_operation_sample.py (hunk 8 above)
Comment (xinyunh0929) [reply jobs/batch_operation_sample.py]: I know. However, as mentioned above, they want half jobs updated using field mask feature, and half not.

**Comment by xinyunh0929** [comment]
How could I add cla:yes back?

File: jobs/api_client/auto_complete_sample.py [hunk 38]
```diff
@@ -0,0 +1,81 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+# [START instantiate]
+from googleapiclient.discovery import build
+
```
Comment (michaelawyu) [inline jobs/api_client/auto_complete_sample.py]: Please remove the extra empty line(s).

Comment (xinyunh0929) [reply jobs/api_client/auto_complete_sample.py]: Done.
